deps/spidershim/src/v8value.cc

```diff
@@ -324,4 +325,15 @@ bool Value::IsNativeError() const {
 }
 
 bool Value::IsExternal() const { return External::IsExternal(this); }
+
+bool
+Value::IsProxy() const
+{
+  JSObject* obj = GetObject(this);
```

Comment:
On Mon, May 30, 2016 at 06:11:18AM -0700, Ehsan Akhgari wrote:
> In deps/spidershim/src/v8value.cc:
> 
> > @@ -324,4 +325,15 @@ bool Value::IsNativeError() const {
> >  }
> >
> >  bool Value::IsExternal() const { return External::IsExternal(this); }
> > +
> > +bool
> > +Value::IsProxy() const
> > +{
> > +  JSObject* obj = GetObject(this);
> 
> You need to check IsObject() first.

GetObject() checks it